### PR TITLE
make port optional to build full url, exclusion rule eval fallback on url path

### PR DESCRIPTION
**Pull request by SrikarMannepalli**
## Description
This PR updates the logic of building the full url to make port optional. It also updates the exclusion rule url filter eval to use url path as fallback in case the full url does not exist.

### Testing
Added unit tests

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 1]
```diff
@@ -283,38 +282,51 @@ private static Optional<String> getHttpUrlForOtelFormatClientSpan(
...
+              .getValue();
+      String url;
+      if (attributeValueMap.containsKey(OTelSpanSemanticConventions.NET_PEER_PORT.getValue())) {
+        url =
+            buildUrl(
+                httpScheme,
+                netPeerIp,
+                attributeValueMap
+                    .get(OTelSpanSemanticConventions.NET_PEER_PORT.getValue())
+                    .getValue(),
```
Comment (skjindal93) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:325]: nit: Can we extract it out in a variable?

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 2]
```diff
@@ -324,32 +336,54 @@ private static Optional<String> getHttpUrlForOtelFormatServerSpan(
...
+      final String httpScheme = getHttpSchemeFromRawAttributes(attributeValueMap).get();
+      final String serverName = attributeValueMap.get(HTTP_SERVER_NAME.getValue()).getValue();
+      final String httpTarget = attributeValueMap.get(HTTP_TARGET.getValue()).getValue();
+      String url;
+      if (attributeValueMap.containsKey(HTTP_NET_HOST_PORT.getValue())) {
+        url =
+            buildUrl(
+                httpScheme,
+                serverName,
+                attributeValueMap.get(HTTP_NET_HOST_PORT.getValue()).getValue(),
```
Comment (skjindal93) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:349]: nit: Can we extract it out in a variable?

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java [hunk 3]
```diff
@@ -324,32 +336,54 @@ private static Optional<String> getHttpUrlForOtelFormatServerSpan(
...
+      final String netHostName = attributeValueMap.get(HTTP_NET_HOST_NAME.getValue()).getValue();
+      final String httpTarget = attributeValueMap.get(HTTP_TARGET.getValue()).getValue();
+
+      String url;
+      if (attributeValueMap.containsKey(HTTP_NET_HOST_PORT.getValue())) {
+        url =
+            buildUrl(
+                httpScheme,
+                netHostName,
+                attributeValueMap.get(HTTP_NET_HOST_PORT.getValue()).getValue(),
```
Comment (skjindal93) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java:368]: nit: Can we extract it out in a variable?

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/ExcludeSpanRuleEvaluator.java [hunk 4]
```diff
@@ -112,29 +112,45 @@ private boolean matchesRelationalSpanFilter(
...
         case FIELD_URL:
-          Optional<String> fullHttpUrlMaybe = HttpSemanticConventionUtils.getFullHttpUrl(event);
-          if (fullHttpUrlMaybe.isEmpty()) {
-            return false;
-          }
-          return spanFilterMatcher.matches(
-              fullHttpUrlMaybe.get(),
-              relationalSpanFilterExpression.getRightOperand(),
-              relationalSpanFilterExpression.getOperator());
+          Optional<String> urlMaybe = getUrl(event);
```
Comment (skjindal93) [inline span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/ExcludeSpanRuleEvaluator.java:124]: `getUrl` doesn't look like it's falling back to HTTP path. Can we instead use `HttpSemanticConventionUtils.getFullHttpUrl(event).or(() ->  HttpSemanticConventionUtils.getHttpPath(event));`

**Review by skjindal93 — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]

**Pull request merged**